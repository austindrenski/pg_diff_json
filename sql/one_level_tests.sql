CREATE EXTENSION pg_diff_json;

-- one-level tests

\echo 'one-level test (identity)'
SELECT diff_jsonb($${
  "a": 0,
  "b": 1
}$$, $${
  "a": 0,
  "b": 1
}$$);

\echo 'one-level test (delete)'
SELECT diff_jsonb($${
  "a": 0,
  "b": 1
}$$, $${
  "a": 0
}$$);

\echo 'one-level test (new value)'
SELECT diff_jsonb($${
  "a": 0,
  "b": 1
}$$, $${
  "a": 0,
  "b": 2
}$$);

\echo 'one-level test (new type)'
SELECT diff_jsonb($${
  "a": 0,
  "b": 1
}$$, $${
  "a": 0,
  "b": true
}$$);

\echo 'one-level test (new value and new type)'
SELECT diff_jsonb($${
  "a": 0,
  "b": 1
}$$, $${
  "a": 1,
  "b": true
}$$);