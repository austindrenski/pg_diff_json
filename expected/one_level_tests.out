CREATE EXTENSION pg_diff_json;

-- one-level tests

\echo 'one-level test (identity)'
one-level test (identity)
SELECT diff_jsonb($${
  "a": 0,
  "b": 1
}$$, $${
  "a": 0,
  "b": 1
}$$);
 diff_jsonb 
------------
 
(1 row)


\echo 'one-level test (delete)'
one-level test (delete)
SELECT diff_jsonb($${
  "a": 0,
  "b": 1
}$$, $${
  "a": 0
}$$);
  diff_jsonb  
--------------
 {"(-,{b},)"}
(1 row)


\echo 'one-level test (new value)'
one-level test (new value)
SELECT diff_jsonb($${
  "a": 0,
  "b": 1
}$$, $${
  "a": 0,
  "b": 2
}$$);
  diff_jsonb   
---------------
 {"(+,{b},2)"}
(1 row)


\echo 'one-level test (new type)'
one-level test (new type)
SELECT diff_jsonb($${
  "a": 0,
  "b": 1
}$$, $${
  "a": 0,
  "b": true
}$$);
    diff_jsonb    
------------------
 {"(+,{b},true)"}
(1 row)


\echo 'one-level test (new value and new type)'
one-level test (new value and new type)
SELECT diff_jsonb($${
  "a": 0,
  "b": 1
}$$, $${
  "a": 1,
  "b": true
}$$);
          diff_jsonb          
------------------------------
 {"(+,{b},true)","(+,{a},1)"}
(1 row)

